bundles/org.openhab.binding.sony/src/main/java/org/openhab/binding/sony/internal/net/SocketChannelSession.java

```diff
@@ -0,0 +1,306 @@
+
+        /**
+         * Runs the logic to read from the socket until interrupted. A 'response' is anything that ends
+         * with a carriage-return/newline combo. Additionally, the special "Login: " and "Password: " prompts are
+         * treated as responses for purposes of logging in.
+         */
+        @Override
+        public void run() {
+            final StringBuilder sb = new StringBuilder(100);
+            final ByteBuffer readBuffer = ByteBuffer.allocate(1024);
+
+            responses.clear();
+
+            while (!Thread.currentThread().isInterrupted()) {
+                try {
+                    // if reader is null, sleep and try again
+                    if (readBuffer == null) {
+                        Thread.sleep(250);
+                        continue;
+                    }
+
+                    final SocketChannel channel = socketChannel.get();
+                    if (channel == null) {
+                        // socket was closed
+                        Thread.currentThread().interrupt();
+                        break;
+                    }
+
+                    final int bytesRead = channel.read(readBuffer);
+                    if (bytesRead == -1) {
+                        responses.put(new IOException("server closed connection"));
+                        break;
+                    } else if (bytesRead == 0) {
+                        readBuffer.clear();
+                        continue;
+                    }
+
+                    readBuffer.flip();
+                    while (readBuffer.hasRemaining()) {
+                        final char ch = (char) readBuffer.get();
+                        sb.append(ch);
+                        if (ch == '\n') {
+                            final String str = sb.toString();
+                            sb.setLength(0);
+                            final String response = str.substring(0, str.length() - 1).trim();
+                            responses.put(response);
+                        }
+                    }
+
+                    readBuffer.flip();
+
+                } catch (final InterruptedException e) {
+                    // Ending thread execution
+                    Thread.currentThread().interrupt();
+                } catch (final AsynchronousCloseException e) {
+                    // socket was closed by another thread but interrupt our loop anyway
+                    Thread.currentThread().interrupt();
+                } catch (final IOException e) {
+                    // set before pushing the response since we'll likely call back our stop
+                    Thread.currentThread().interrupt();
+
+                    try {
+                        responses.put(e);
+                        break;
+                    } catch (final InterruptedException e1) {
+                        // Do nothing - probably shutting down
+                        // Since we set isRunning to false, will drop out of loop and end the thread
```

Comment:
Looks like this assumption is no longer true. I would suggest resetting the interrupt flag to make sure it exits.

Reply:
done